std-bits/base/src/main/java/org/enso/base/Text_Utils.java

```diff
@@ -162,29 +158,83 @@ public static int[] get_codepoints(String str) {
    * @return array of substrings of {@code str} contained between occurences of {@code sep}
    */
   public static String[] split_by_literal(String str, String sep) {
-    return str.split(Pattern.quote(sep));
+    return str.split(Pattern.quote(sep), -1);
   }
 
   /**
    * Splits the string on each occurrence of UTF-8 whitespace, returning the resulting substrings in
    * an array.
    *
    * @param str the string to split
+   * @param keep_whitespace whether to keep the whitespace as separate entries
    * @return the array of substrings of {@code str}
    */
-  public static String[] split_on_whitespace(String str) {
-    return whitespace.split(str);
+  public static List<String> split_on_whitespace(String str, boolean keep_whitespace) {
```

Comment:
as far as I can see this is now just for a Benchmark so shall we remove and tidy up?